src/main/java/org/openrewrite/staticanalysis/CatchClauseOnlyRethrows.java

```diff
@@ -110,11 +110,6 @@ private boolean onlyRethrows(J.Try.Catch aCatch) {
                 }
 
                 Expression exception = ((J.Throw) aCatch.getBody().getStatements().get(0)).getException();
-                JavaType.FullyQualified catchType = TypeUtils.asFullyQualified(aCatch.getParameter().getType());
-                if (catchType == null || !catchType.equals(exception.getType())) {
-                    return false;
-                }
```

Comment:
Instead of removing this check, I wonder if we should add an additional check with something like `if (catchType instanceof JavaType.MultiCatch) { ... check if any of the thrown exceptions matches ... }`. I'd feel slightly more secure with such an approach then dropping this requirement.

Reply:
Hi Tim, Thanks for the review! I am not sure how to implement this check.... 

In the base case e.g. `catch(A ex) {throw ex}` type A is assigned to `ex` in the body. Hence we can compare the type of the catch parameter `ex` that of the body. In the multicatch, `catch (A | B ex) {throw ex}` both `ex` in the body and parameter will be assigned type `Exception`, as it is not know if they will be of type A or B. Using `JavaType.MultiCatch.getThrowableTypes` I can retrieve the possible types the parameter `ex` can take, but the I can't compare them with the type of `ex` in the body as this will still be of type `Exception`.

Reply:
If we can't determine it based on type then maybe in case of a `MultiCatch` we could check whether `exception` is an identifier and if it is whether it's the same identifier as `aCatch.getParameter()`?

Reply:
Isn't that already checked on line 114?

```java
if (exception instanceof J.Identifier) {
    return ((J.Identifier) exception).getSimpleName().equals(aCatch.getParameter().getTree().getVariables().get(0).getSimpleName());
}
```

Reply:
Ah yes, missed that part! Though checking by name is potentially more dangerous than checking by reference.
Not sure what direction to go here then